### PR TITLE
LPS-143420

### PR DESCRIPTION
**Pull request by julschong**
Hi Tina, @tinatian
this is resubmitted to separate by each **Util classes

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test

File: portal-impl/portal-impl.iml [hunk 1]
```diff
@@ -1,68 +1,73 @@
...
-		<orderEntry type="module" module-name="util-java" />
-		<orderEntry type="module" module-name="util-taglib" />
-		<orderEntry type="module" module-name="portal-kernel" scope="PROVIDED" />
-		<orderEntry type="module" module-name="portal-test" scope="TEST" />
-		<orderEntry type="module" module-name="portal-test-integration" scope="TEST" />
-		<orderEntry type="library" scope="PROVIDED" name="development" level="project" />
-		<orderEntry type="library" name="portal" level="project" />
-		<orderEntry type="library" scope="PROVIDED" name="global" level="project" />
-	</component>
+  <component name="FacetManager">
```
Comment (tinatian) [inline portal-impl/portal-impl.iml:3]: @julschong This portal-impl.iml seems unrelated to other changes in the same commit.

**Comment by tinatian** [comment]
@julschong There are almost 30 commits in this pull, could you please try to separate them into different pulls? Two reasons:
1. Less commits are easier to review, I need more time to review a pull with 30 commits so I have to find a big slot of time for it which may delay the review if I could not find. But if the commit is around 10 commits, it would be easier for me, I may finish the review when I have a few mins.
2. Less commits are easier to troubleshot when there is an issue. Suppose QA reported an regression on your changes after they are merged, QA will usually tell you that your commits for a ticket is the cause, and it is your responsibility to figure out which commit is the root cause. Less commits will let you figure out the cause quickly.

Please see my comments and resend.

Thanks.
Tina.

**Comment by julschong** [comment]
I'll separate this pull and send with less commits again
